jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -1039,10 +1039,11 @@ private static final Object resolveConstantDynamic(long j9class, String name, St
 				break;
 			case 3: {
 				int cpValue = cp.getIntAt(index);
-				Class<?> argClass;
+				Class<?> argClass = null;
 				if (treatLastArgAsVarargs && (staticArgIndex >= (bsmTypeArgCount - 1))) {
 					argClass = varargsComponentType;
-				} else {
+				/* verify that a call to parameterType will not cause an ArrayIndexOutOfBoundsException */
+				} else if (bsmTypeArgCount > staticArgIndex) {
```

Comment:
Can you modify that to be `staticArgIndex < bsmTypeArgCount`?

Reply:
This is also kind of subtle - it fails to set argClass and counts on getting the WMTE later.  Can you add a comment that explains this behaviour?  It'll make it clearer if we have to look at this again

Reply:
updated.